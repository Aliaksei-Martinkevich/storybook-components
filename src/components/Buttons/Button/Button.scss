.button {
    box-sizing: border-box;
    font-size: 14px;
    font-family: 'Oxygen Regular';
    line-height: 18px;
    height: 42px; 
    border-radius: 22px;
    padding-right: 44px;
    padding-left: 44px;

    &:focus {
        outline: 0;
    };

    &::-moz-focus-inner {
        border: 0;
    };
}