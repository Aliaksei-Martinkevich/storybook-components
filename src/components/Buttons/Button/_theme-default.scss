@import '../../../colors.scss';

$background-regular: $white;
$background-hover: $white;
$background-selected: $white;
$background-disabled: $light_gray;
$font-regular: $text;
$font-disabled: $gray_secondary_btn_pressed;
$background-processing-start: $background-selected;
$background-processing-end: $background-regular;
$border-regular: $black;
$border-disabled: $gray_secondary_btn_pressed;
$border-width: 1px;
$border-type: solid;