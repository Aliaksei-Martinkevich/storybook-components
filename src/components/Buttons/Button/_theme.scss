.button {
    color: $font-regular;
    background-color: $background-regular;
    border: $border-width $border-type $border-regular;
}

.button:hover {
    background-color: $background-hover;
}

.button:active {
    background-color: $background-selected;
}

.button:disabled {
    background-color: $background-disabled;
    color: $font-disabled;
    border-color: $border-disabled;
}

.button_processing {
    background: $background-processing-end;
    background: linear-gradient(to right, $background-processing-start 0%, $background-processing-start 50%, $background-processing-end 50%, $background-processing-end 100%);
}