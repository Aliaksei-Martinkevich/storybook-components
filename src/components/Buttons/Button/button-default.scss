@import '_theme-default';
@import '_theme';