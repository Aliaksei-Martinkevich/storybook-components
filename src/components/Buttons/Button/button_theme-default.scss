@import '../../../../assets/scsss/colors';
@import 'theme';

$background-regular: $white;
$background-hover: $white;
$background-selected: $white;
$background-disabled: $light_gray;
$font-regular: $text;
$font-disabled: $gray_secondary_btn_pressed;
$border-regular: $black;
$border-disabled: $gray_secondary_btn_pressed;
$border-width: 1px;
$border-type: solid;

.button_theme-default {
    border-width: $border-width;
    border-style: $border-type;

    @include colors($font-regular, $background-regular, $border-regular);

    &:hover { background-color: $background-hover; }

    &:active { background-color: $background-selected; }

    &:disabled { @include colors($font-disabled, $background-disabled, $border-disabled); }

    &.button_processing { @include two-colors-gradient($background-selected, $background-regular); }
}