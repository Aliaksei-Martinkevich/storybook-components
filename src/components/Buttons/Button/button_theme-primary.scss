@import '../../../../assets/scsss/colors';
@import 'theme';

$background-regular: $salmon;
$background-hover: $salmon_primary_btn_hover;
$background-selected: $salmon_primary_btn_selected;
$background-disabled: $light_gray;
$font-regular: $white;
$font-disabled: $gray;
$border-regular: $black;
$border-disabled: $light_gray;
$border-width: 0px;
$border-type: solid;

.button_theme-primary {
    border-width: $border-width;
    border-style: $border-type;

    @include colors($font-regular, $background-regular, $border-regular);

    &:hover { background-color: $background-hover; }

    &:active { background-color: $background-selected; }

    &:disabled { @include colors($font-disabled, $background-disabled, $border-disabled); }

    &.button_processing:disabled { color: $font-regular; @include two-colors-gradient($background-selected, $background-regular); }
}