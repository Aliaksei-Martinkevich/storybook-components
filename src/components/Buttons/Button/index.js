import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';
import defaultSettings from './settings.scss';

import ButtonContent from '../ButtonContent';

const Button = ({
  text,
  onClick,
  disabled = false,
  processing = false,
  iconUri = undefined,
  style = undefined,
  className = '',
}) => {
  let settings = defaultSettings;
  if (style) {  
    settings = style;
  }
  
  return (
      <button style={settings}
              className={`button ${processing ? 'processing ' : ''} ${className}`}
              onClick={onClick}
              disabled={disabled}
              title={text}>
        
        <ButtonContent iconUri={processing ? iconUri : undefined} text={text}/>
       
      </button>
  );
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  processing: PropTypes.bool,
}

export default Button;
