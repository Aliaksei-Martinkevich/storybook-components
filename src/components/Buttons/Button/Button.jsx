import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './button_theme-default.scss';
import './button_theme-primary.scss';
import './button_theme-secondary.scss';
import './button_theme-transparent.scss';
import './Button.scss';
import ButtonContent from './ButtonContent/ButtonContex.jsx';

export const BUTTON_THEMES = {
  DEFAULT: 'button_theme-default',
  PRIMARY: 'button_theme-primary',
  SECONDARY: 'button_theme-secondary',
  TRANSPARENT: 'button_theme-transparent',
};

const Button = ({ text, onClick, disabled, processing, iconUri, className, theme }) => (
  <button
    className={classNames('button', theme, { button_processing: processing }, className)}
    onClick={onClick}
    disabled={disabled || processing}
    title={text}
  >
    <ButtonContent iconUri={processing && iconUri} text={text} />
  </button>
  );

Button.defaultProps = {
  onClick: undefined,
  disabled: false,
  processing: false,
  iconUri: '',
  className: '',
  theme: BUTTON_THEMES.DEFAULT,
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  processing: PropTypes.bool,
  iconUri: PropTypes.string,
  className: PropTypes.string,
  theme: PropTypes.oneOf(Object
      .getOwnPropertyNames(BUTTON_THEMES)
      .map(name => BUTTON_THEMES[name])),
};

export default Button;
