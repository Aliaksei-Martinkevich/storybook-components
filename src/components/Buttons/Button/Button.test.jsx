import React from 'react';
import renderer from 'react-test-renderer';
import Button, { BUTTON_THEMES } from './Button.jsx';
import icon from '../../../../assets/icons/svg/loading-circle-dashes-black.svg';

it('Should render without fails', () => {
  const customClassName = 'class-name-example';
  const text = 'Some text';
  const component = renderer.create(
    <Button
      className={customClassName}
      text={text}
      theme={BUTTON_THEMES.DEFAULT}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Should add processing class', () => {
  const customClassName = 'class-name-example';
  const text = 'Some text';
  const component = renderer.create(
    <Button
      className={customClassName}
      text={text}
      iconUri={icon}
      processing
      theme={BUTTON_THEMES.DEFAULT}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
