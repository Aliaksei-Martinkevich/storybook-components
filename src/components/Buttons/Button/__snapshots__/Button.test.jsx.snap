// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should add processing class 1`] = `
<button
  className="button button_theme-default button_processing class-name-example"
  disabled={true}
  onClick={undefined}
  title="Some text"
>
  <div
    className="button-content"
  >
    <img
      alt=""
      className="load-icon button-content__icon"
      src="test-file-stub"
    />
    <span
      className="button-label butten-content__label"
    >
      Some text
    </span>
  </div>
</button>
`;

exports[`Should render without fails 1`] = `
<button
  className="button button_theme-default class-name-example"
  disabled={false}
  onClick={undefined}
  title="Some text"
>
  <div
    className="button-content"
  >
    <span
      className="button-label butten-content__label"
    >
      Some text
    </span>
  </div>
</button>
`;
